### ISSUE TITLE
Plugin throwing error on arbitrary spacing values when spacing values are defined in @theme?

### ISSUE THREAD

**Issue by vincerubinetti** [schoero/eslint-plugin-better-tailwindcss#329]
### Description

I'm experimenting with only allowing certain spacings in my `@theme` to enforce more visual consistency:

```css
@theme {
  --spacing-*: initial;
  --spacing-1: 0.25rem;
  --spacing-2: 0.5rem;
  --spacing-4: 1rem;
  --spacing-8: 2rem;
  --spacing-12: 3rem;
  --spacing-16: 4rem;
  --spacing-20: 5rem;
  --spacing-24: 6rem;
}
```

But when I added this, this plugin starts to get tripped up by arbitrary spacing values in brackets, e.g. `px-[calc(100%+--spacing(2))]`.

I don't think `-*: initial` is supposed to prevent/unset arbitrary values for things, only the enumerated values.

See the attached ESLint log. Also if I disable the plugin, the console stops showing errors and is able to lint the a file again.

### Flavor

TSX

### Code Input

```tsx
`px-[max(var(--pad),(100%---spacing(1)*var(--width))/2)]`
```

### Expected Behavior

```tsx
Plugin is able to parse the arbitrary `px` (or any other similar) value without erroring.
```

### Actual Behavior

```tsx
The plugin errors.
```

### ESLint Log

```shell
2026-02-20 23:13:26.963 [error] 2026-02-21T04:13:26.962Z eslint:eslint Using config loader LegacyConfigLoader
2026-02-20 23:13:26.963 [error] 2026-02-21T04:13:26.962Z eslint:config-loader Calculating config for file /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.963 [error] 2026-02-21T04:13:26.962Z eslint:config-loader Searching for eslint.config.js
2026-02-20 23:13:26.963 [error] 2026-02-21T04:13:26.962Z eslint:config-loader [Legacy]: Calculating config for /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.963 [error] 2026-02-21T04:13:26.962Z eslint:config-loader [Legacy]: Using config file /Users/vincerubinetti/Desktop/xxx.com/eslint.config.js and base path /Users/vincerubinetti/Desktop/xxx.com
2026-02-20 23:13:26.964 [error] 2026-02-21T04:13:26.962Z eslint:config-loader Calculating config array from config file /Users/vincerubinetti/Desktop/xxx.com/eslint.config.js and base path /Users/vincerubinetti/Desktop/xxx.com
2026-02-20 23:13:26.964 [error] 2026-02-21T04:13:26.963Z eslint:config-loader Loading config file /Users/vincerubinetti/Desktop/xxx.com/eslint.config.js
2026-02-20 23:13:26.964 [error] 2026-02-21T04:13:26.963Z eslint:config-loader Loading config from /Users/vincerubinetti/Desktop/xxx.com/eslint.config.js
2026-02-20 23:13:26.964 [error] 2026-02-21T04:13:26.963Z eslint:config-loader Config file URL is file:///Users/vincerubinetti/Desktop/xxx.com/eslint.config.js
2026-02-20 23:13:26.964 [error] 2026-02-21T04:13:26.963Z eslint:config-loader Config file /Users/vincerubinetti/Desktop/xxx.com/eslint.config.js is not empty
2026-02-20 23:13:26.974 [error] 2026-02-21T04:13:26.974Z eslint:linter Linting code for /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx (pass 1)
2026-02-20 23:13:26.975 [error] 2026-02-21T04:13:26.974Z eslint:linter Verify
2026-02-20 23:13:26.975 [error] 2026-02-21T04:13:26.974Z eslint:linter With flat config: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.975 [error] 2026-02-21T04:13:26.974Z eslint:languages:js Parsing: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.976 [error] 2026-02-21T04:13:26.976Z eslint:languages:js Parsing successful: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.976 [error] 2026-02-21T04:13:26.976Z eslint:languages:js Scope analysis: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.976 [error] 2026-02-21T04:13:26.976Z eslint:languages:js Scope analysis successful: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.988 [error] 2026-02-21T04:13:26.987Z eslint:linter An error occurred while traversing
2026-02-20 23:13:26.988 [error] 2026-02-21T04:13:26.988Z eslint:linter Filename: /Users/vincerubinetti/Desktop/xxx.com/app/pages/extras/Blog.tsx
2026-02-20 23:13:26.988 [error] 2026-02-21T04:13:26.988Z eslint:linter Line: 8
2026-02-20 23:13:26.988 [error] 2026-02-21T04:13:26.988Z eslint:linter Parser Options: { ecmaFeatures: { jsx: true } }
2026-02-20 23:13:26.988 [error] 2026-02-21T04:13:26.988Z eslint:linter Settings: { 'better-tailwindcss': { entryPoint: 'app/styles.css' } }
2026-02-20 23:13:26.988 [error] An unexpected error occurred:
2026-02-20 23:13:26.988 [error] Error: Cannot apply unknown utility class `px-[max(var(--pad),(100%---spacing(1)*var(--width))/2)]`
    at onInvalidCandidate (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:21:1511)
    at $e (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:16:37066)
    at file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:21:355
    at Ar (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:3:1733)
    at I (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:3:1377)
    at Ce (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:21:172)
    at Gi (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:38:294)
    at async wf (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/tailwindcss/dist/chunk-O3SHLFNR.mjs:38:1577)
    at async file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/eslint-plugin-better-tailwindcss/lib/tailwindcss/context.async.v4.js:20:20
    at async Object.getClassOrder (file:///Users/vincerubinetti/Desktop/xxx.com/node_modules/eslint-plugin-better-tailwindcss/lib/tailwindcss/tailwind.async.worker.v4.js:17:33)
```

### ESLint Config

```typescript
import eslintJs from "@eslint/js";
import eslintPluginBetterTailwindcss from "eslint-plugin-better-tailwindcss";
import eslintPluginJsxA11y from "eslint-plugin-jsx-a11y";
import * as eslintPluginMdx from "eslint-plugin-mdx";
import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
import eslintPluginReactHooks from "eslint-plugin-react-hooks";
import { defineConfig, globalIgnores } from "eslint/config";
import globals from "globals";
import typescriptEslint from "typescript-eslint";

export default defineConfig([
  globalIgnores([
    "build",
    "public",
    ".react-router",
    "playwright-report",
    "test-results",
  ]),

  // https://github.com/mdx-js/eslint-mdx/issues/92
  {
    name: "TypeScript",
    extends: typescriptEslint.configs.recommended,
    ignores: ["**/*.mdx"],
    rules: {
      "@typescript-eslint/no-unused-vars": ["warn", { caughtErrors: "none" }],
      "@typescript-eslint/consistent-type-definitions": ["error", "type"],
      "@typescript-eslint/consistent-type-imports": "error",
    },
  },
  {
    name: "JavaScript",
    ...eslintJs.configs.recommended,
    rules: {
      "prefer-const": ["error", { destructuring: "all" }],
    },
  },
  {
    name: "React Hooks",
    ...eslintPluginReactHooks.configs.flat.recommended,
  },
  {
    name: "JSX Accessibility",
    ...eslintPluginJsxA11y.flatConfigs.recommended,
    rules: {
      // https://github.com/dequelabs/axe-core/issues/4566
      "jsx-a11y/no-noninteractive-tabindex": ["off"],
    },
  },
  {
    name: "Prettier",
    ...eslintPluginPrettierRecommended,
    ignores: ["**/*.mdx"],
  },
  {
    name: "Tailwind",
    extends: [eslintPluginBetterTailwindcss.configs.recommended],
    rules: {
      "better-tailwindcss/enforce-consistent-line-wrapping": [
        "warn",
        {
          preferSingleLine: true,
          group: "never",
          printWidth: 0,
        },
      ],
      "better-tailwindcss/no-unknown-classes": ["warn", { ignore: ["dark"] }],
    },
    settings: {
      "better-tailwindcss": { entryPoint: "app/styles.css" },
    },
  },
  {
    name: "MDX",
    ...eslintPluginMdx.flat,
  },
  {
    languageOptions: {
      globals: globals.browser,
      ecmaVersion: 2020,
    },
  },
]);
```

### Versions

- ESLint: 9.39.2
- Parser: 
- Plugin: 4.1.1
- Node: 22
